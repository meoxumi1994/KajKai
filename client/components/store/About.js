import React from 'react'

import ShowInMap from '~/containers/entity/map/ShowInMap'

const About = ({ category, firstCategory, secondCategory, urlname, phone, storename, numfollow, numlike,  address, position }) => {
    return(
        <div style={{
            borderRadius: 4,
            border: '1px solid #CCCCCC',
            boxShadow: '0px 0px 4px #CCCCCC',
            backgroundColor: 'white',
            width: 938, marginTop: 10, marginLeft: -23,
            }}>
            <div className="container-fluid" style={{ padding: 10 }}>
                <div className="row">
                    <div className="col col-xs-2">
                        <span style={{ fontWeight: 'bold' }}>User name </span>
                    </div>
                    <div className="col col-xs-9" style={{ marginLeft: 28 }}>
                        <span>{storename}</span>
                    </div>
                </div>
            </div>

            <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>
            <div className="container-fluid" style={{ padding: 10 }}>
                <div className="row">
                    <div className="col col-xs-2">
                        <span style={{ fontWeight: 'bold' }}>Category 1</span>
                    </div>
                    <div className="col col-xs-9" style={{ marginLeft: 28 }}>
                        <span>{firstCategory}</span>
                    </div>
                </div>
            </div>
            <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>

            <div className="container-fluid" style={{ padding: 10 }}>
                <div className="row">
                    <div className="col col-xs-2">
                        <span style={{ fontWeight: 'bold' }}>Category 2</span>
                    </div>
                    <div className="col col-xs-9" style={{ marginLeft: 28 }}>
                        <span>{secondCategory}</span>
                    </div>
                </div>
            </div>
            <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>

            <div className="container-fluid" style={{ padding: 10 }}>
                <div className="row">
                    <div className="col col-xs-2">
                        <span style={{ fontWeight: 'bold' }}>Category</span>
                    </div>
                    <div className="col col-xs-9" style={{ marginLeft: 28 }}>
                        <span>{category}</span>
                    </div>
                </div>
            </div>
            <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>

            <div className="container-fluid" style={{ padding: 10 }}>
                <div className="row">
                    <div className="col col-xs-2">
                        <span style={{ fontWeight: 'bold' }}>Phone</span>
                    </div>
                    <div className="col col-xs-9" style={{ marginLeft: 28 }}>
                        <span>{phone}</span>
                    </div>
                </div>
            </div>
            <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>

            <div className="container-fluid" style={{ padding: 10 }}>
                <div className="row">
                    <div className="col col-xs-2">
                        <span style={{ fontWeight: 'bold' }}>numfollow</span>
                    </div>
                    <div className="col col-xs-9" style={{ marginLeft: 28 }}>
                        <span>{numfollow}</span>
                    </div>
                </div>
            </div>
            <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>

            <div className="container-fluid" style={{ padding: 10 }}>
                <div className="row">
                    <div className="col col-xs-2">
                        <span style={{ fontWeight: 'bold' }}>numlike</span>
                    </div>
                    <div className="col col-xs-9" style={{ marginLeft: 28 }}>
                        <span>{numlike}</span>
                    </div>
                </div>
            </div>
            <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>

            <div className="container-fluid" style={{ padding: 10 }}>
                <div className="row">
                    <div className="col col-xs-2">
                        <span style={{ fontWeight: 'bold' }}>address</span>
                    </div>
                    <div className="col col-xs-9" style={{ marginLeft: 28 }}>
                        <span>{address}</span>
                    </div>
                </div>
            </div>
            <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>

            <ShowInMap width={935} height={400} position={position}/>
        </div>
    )
}

export default About
