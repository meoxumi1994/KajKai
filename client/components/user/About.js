import React from 'react'

import AboutCell from '~/containers/entity/row/AboutCell'

import { getTime } from '~/containers/support'

const getLongLanguage = (language) => {
    if(language == 'vi'){
        return 'Viá»‡t Nam'
    }
    if(language == 'en'){
        return 'English'
    }
    return ""
}

class About extends React.Component {
    constructor(props){
        super(props)
        this.state = {
            show : false,
        }
    }
    _onClick(){
        this.setState({ show: !this.state.show })
    }
    render(){
        const { username, email, language, address, phone, age , isOwner, position, interactive,
            USER_NAME, EMAIL, LANGUAGE, ADDRESS, PHONE, AGE, INFO_GENERAL, POSITION_IN_MAP, INTERACTION,
            TOTAL_LIKE, TOTAL_COMMENT, TOTAL_REPLY_COMMENT, TOTAL_FOLLOW, CREATE_TIME, LAST_TIME } = this.props
        return(
            <div>
                <div style={{
                    borderRadius: 4,
                    border: '1px solid #DFE0E4',
                    backgroundColor: 'white',
                    width: 788, marginTop: 10,
                    }}>
                    {this.state.show && 'hihihi'}
                    <div style={{ fontSize: 20, padding: 10, backgroundColor: '#F6F7F9', borderRadius: '5px 5px 0px 0px'}}>
                        <img src={"/images/inforicon.svg"} width={20} height={20}/>
                        <span style={{ marginLeft: 10 }}>{INFO_GENERAL}</span>
                    </div>
                    <hr style={{ margin: 0, padding: 0 }}/>
                    <AboutCell kind="username" title={USER_NAME} value={username}/>
                    <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>
                    <AboutCell kind="email" title={EMAIL} value={email}/>
                    <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>
                    <AboutCell kind="language" title={LANGUAGE} value={getLongLanguage(language)}/>
                    <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>
                    <AboutCell kind="address" title={ADDRESS} value={address}/>
                    <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>
                    <AboutCell kind="phone" title={PHONE} value={phone}/>
                    <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>
                    <AboutCell kind="age" title={AGE} value={age}/>
                    <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>
                    <AboutCell kind="position" title={POSITION_IN_MAP} value={position}/>
                </div>
                <div style={{
                    borderRadius: 4,
                    border: '1px solid #DFE0E4',
                    backgroundColor: 'white',
                    width: 788, marginTop: 10,
                    }}>
                    {this.state.show && 'hihihi'}
                    <div style={{ fontSize: 20, padding: 10, backgroundColor: '#F6F7F9', borderRadius: '5px 5px 0px 0px'}}>
                        <img src={"/images/interactiveicon.svg"} width={22} height={22}/>
                        <span style={{ marginLeft: 10 }}>{INTERACTION}</span>
                    </div>
                    <hr style={{ margin: 0, padding: 0 }}/>
                    {interactive &&
                        <div>
                            <AboutCell kind="leadercomment" title={TOTAL_COMMENT} value={interactive.numleadercomment}/>
                            <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>
                            <AboutCell kind="comment" title={TOTAL_REPLY_COMMENT} value={interactive.numcomment}/>
                            <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>
                            <AboutCell kind="like" title={TOTAL_LIKE} value={interactive.numlike}/>
                            <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>
                            <AboutCell kind="follow" title={TOTAL_FOLLOW} value={interactive.numfollow}/>
                            <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>
                            <AboutCell kind="create_time" title={CREATE_TIME} value={getTime(interactive.create_time)}/>
                            <hr style={{ margin: 0, padding: 0, borderColor: '#E9EbEE'}}/>
                            <AboutCell kind="last_time" title={LAST_TIME} value={getTime(interactive.last_time)}/>
                        </div>
                    }
                </div>
            </div>

        )
    }
}
export default About
