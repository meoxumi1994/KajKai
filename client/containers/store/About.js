import { connect } from 'react-redux'
import { get } from '~/config/allString'

import About from '~/components/store/About'

const mapStateToProps = (state, { id }) => {
    const g = (lang) => get(state.user.language, lang)
    let isOwner = state.user.id == state.inst.user.index.id
    return({
        ...state.user,
        ...state.inst.store.index,
        isOwner: isOwner,
        USER_NAME: g('USER_NAME'),
        EMAIL: g('EMAIL'),
        LANGUAGE: g('LANGUAGE'),
        ADDRESS: g('ADDRESS'),
        PHONE: g('PHONE'),
        AGE: g('AGE'),
        INFO_GENERAL: g('INFO_GENERAL'),
        POSITION_IN_MAP: g('POSITION_IN_MAP'),
        INTERACTION: g('INTERACTION'),
        TOTAL_COMMENT: g('TOTAL_COMMENT'),
        TOTAL_REPLY_COMMENT: g('TOTAL_REPLY_COMMENT'),
        TOTAL_LIKE: g('TOTAL_LIKE'),
        TOTAL_FOLLOW: g('TOTAL_FOLLOW'),
        CREATE_TIME: g('CREATE_TIME'),
        LAST_TIME: g('LAST_TIME'),
        CATEGORY: g('CATEGORY'),
        URL_NAME: g('URL_NAME'),
        STORE_NAME: g('STORE_NAME'),
        categories: state.inst.category,
    })
}

const mapDispatchToProps = (dispatch, ownProps) => ({

})

const AboutContainer = connect(
    mapStateToProps, mapDispatchToProps
)(About)

export default AboutContainer
