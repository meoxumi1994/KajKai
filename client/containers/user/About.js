import { connect } from 'react-redux'
import { get } from '~/config/allString'

import About from '~/components/user/About'

const mapStateToProps = (state, ownProps) => {
    const g = (lang) => get(state.user.language, lang)
    return({
    })
}

const mapDispatchToProps = (dispatch, ownProps) => ({

})

const AboutContainer = connect(
    mapStateToProps, mapDispatchToProps
)(About)

export default AboutContainer
