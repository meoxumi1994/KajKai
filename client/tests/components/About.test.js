import React from 'react'
import About from '../../components/user/About'
import renderer  from 'react-test-renderer'
import {shallow} from 'enzyme';

describe('About', () => {
  it('should render self and subcomponents', () => {
    const component = shallow(
      <About/>
    )

    const clickme = component.find('div.btn.btn-default')
    expect(clickme.text()).toEqual('click me')
    clickme.simulate('click')

    expect(component.text()).toMatch(/hihihi/)

    clickme.simulate('click')
    expect(component.text()).not.toMatch(/hihihi/)

  })
})
