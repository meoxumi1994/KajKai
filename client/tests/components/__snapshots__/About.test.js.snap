// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should render self and subcomponents 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "border": "1px solid #CCCCCC",
      "borderRadius": 4,
      "boxShadow": "0px 0px 4px #CCCCCC",
      "marginLeft": -23,
      "marginTop": 10,
      "width": 838,
    }
  }
>
  <div
    className="btn btn-default"
    onClick={[Function]}
  >
    click me
  </div>
  <div
    style={
      Object {
        "backgroundColor": "#F6F7F9",
        "fontSize": 20,
        "padding": 10,
      }
    }
  >
    About
  </div>
  <hr
    style={
      Object {
        "margin": 0,
        "padding": 0,
      }
    }
  />
  <div
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <span
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      User name : 
    </span>
    <span />
  </div>
  <hr
    style={
      Object {
        "margin": 0,
        "padding": 0,
      }
    }
  />
  <div
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <span
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Email : 
    </span>
    <span />
  </div>
  <hr
    style={
      Object {
        "margin": 0,
        "padding": 0,
      }
    }
  />
  <div
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <span
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Language : 
    </span>
    <span>
      vietnam
    </span>
  </div>
  <hr
    style={
      Object {
        "margin": 0,
        "padding": 0,
      }
    }
  />
  <div
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <span
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Address : 
    </span>
    <span />
  </div>
  <hr
    style={
      Object {
        "margin": 0,
        "padding": 0,
      }
    }
  />
  <div
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <span
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      Age : 
    </span>
    <span />
  </div>
  <hr
    style={
      Object {
        "margin": 0,
        "padding": 0,
      }
    }
  />
</div>
`;
